### ISSUE TITLE
[Feature] ua overrides for Web.Simple

### ISSUE THREAD

**Issue by rijnhard** [onsip/SIP.js#586]
**Is your feature request related to a problem? Please describe.**
It would be very useful to be able to provide more advanced params for `SIP.Web.Simple`, I know it sounds counter-intuitive but I assume the reason for `SIP.Web.Simple` is to give a batteries-included option, that wraps dom handling, event listening and init. The downside is that it is very very limiting with constructor params (obviously because it uses its own scheme to make life easier) but I see no reason for this to be the case. 

**Describe the solution you'd like**
We can use all the goodness of the batteries included option, with the added flexibility of more constructor params by providing an overrides option.

```javascript
// usage
new SIP.Web.Simple(configuration, uaOverrides);

// definition
var Simple = function (options, uaOverrides) {
	// ...
    this.ua = new SIP.UA(Object.assign({
        // User Configurable Options
        uri:               this.options.ua.uri,
        authorizationUser: this.options.ua.authorizationUser,
        password:          this.options.ua.password,
        displayName:       this.options.ua.displayName,
        // Undocumented "Advanced" Options
        userAgentString:   this.options.ua.userAgentString,
        // Fixed Options
        register:          true,
        sessionDescriptionHandlerFactoryOptions: sessionDescriptionHandlerFactoryOptions,
        transportOptions: {
          traceSip: this.options.ua.traceSip,
          wsServers: this.options.ua.wsServers
        }
    }, uaOverrides || {});

	// ...
};
```

This allows all the goodness of the batteries included approach and the flexibility for more advanced use cases, which effectively bridges the two API's.

**Comment by rijnhard** [comment]
Just a note: `Object.assign` is probably not a good idea, and a deep merge is probably a better idea.
Either that or special case handling for `transportOptions`

**Comment by egreenmachine** [comment]
We have no interest in dealing with media within SIP.js. It really is an application level thing. When we made the move to break out of dealing with media internally in 0.8.0, the Simple layer was a compromise to help get new users up and running quickly. The idea is that you can make a basic call and do basic things, but if you want to do anything advanced you should move to the full api, and if you want a truly great user experience you will create your own session description handler. We could have allowed full UA options but purposely chose not to in an attempt to force only the most basic uses to use Simple and have everyone else continue to use the full API.

All that being said, I am really not sure what there is to gain here. If we wanted to move SIP.js towards the Simple interface this would be a great stepping stone. However, we really just want to keep Simple around to help get users up and running quickly and that is it.

**Comment by rijnhard** [comment]
I can understand that, the library is being positioned to deal with a particular problem only (which follows node philosophy).

Essentially this would mean that Teleforge (the company I work for) would need to wrap SIP.js and provide that to our partners where previously they could use `sip.js` directly with a bit of guidance on input params.

Honestly, I'd say Simple should be its own library, which has its own raison d'etre, and is simply built on top of `sip.js`. Then this feature request is still valid. 
Whether you are willing to maintain that is a different question. (heck at this rate I can justify us maintaining it.)

tl;dr if you provide simple IMHO you should do this, although it does make more sense to make simple its own lib, which would help with the ned goal of not doing it at all in `sip.js`

**Comment by egreenmachine** [comment]
As I mentioned earlier, we will not be making the full UA options configurable from `Simple`. You have essentially said that you can solve the problem the same way twice, which is wrap SIP.js in either your own library or break out Simple into it's own project and do what you need and maintain it for your needs. This is pretty much the solution we are hoping people will implement. Use SIP.js for SIP signaling and handle the media on your own or use some other library to handle the media. I am going to mark this as `wontfix` and close it. Thanks
